plan/task.go

```diff
@@ -339,6 +339,11 @@ func (p *TopN) attach2Task(tasks ...task) task {
 	// This is a topN plan.
 	if copTask, ok := t.(*copTask); ok && p.canPushDown() {
 		pushedDownTopN := p.Copy().(*TopN)
+		newByItems := make([]*ByItems, 0, len(p.ByItems))
+		for _, expr := range p.ByItems {
+			newByItems = append(newByItems, expr.Clone())
+		}
+		pushedDownTopN.ByItems = newByItems
```

Comment:
I think this logic should be implemented in `TopN.Copy` method.

Reply:
But there's cases that `TopN.Copy()` don't need to make a copy of its ByItems, such as line 367 of `task.go`,

Reply:
OK, It seems like all the Copy methods just do a shallow copy.